### DISCUSSION TITLE
Does v4 extend or modify Bootstrap's css?

### DISCUSSION THREAD

**Discussion by ZhangChengLin** [ColorlibHQ/AdminLTE#3715] [Q&A]
Can AdminLTE v4 retain all Bootstrap css settings?
Is it based on Bootstrap to expand and increase according to the needs of AdminLTE, without modifying the original settings of Bootstrap's css?

**Comment by ZhangChengLin** [comment]
> https://github.com/twbs/bootstrap/blob/main/site/content/docs/5.0/migration.md#javascript
> Breaking Data attributes for all JavaScript plugins are now namespaced to help distinguish Bootstrap functionality from third parties and your own code. For example, we use data-bs-toggle instead of data-toggle.


Should AdminLTE consider carrying out its own namespace at the same time.
such as:

https://github.com/ColorlibHQ/AdminLTE/blob/a416320ccf137b5f842afe7babfb69b46d620d24/src/ts/treeview.ts#L22

https://github.com/ColorlibHQ/AdminLTE/blob/a416320ccf137b5f842afe7babfb69b46d620d24/src/ts/push-menu.ts#L32-L34

**Reply by ZhangChengLin** [reply]
> https://github.com/ColorlibHQ/AdminLTE/blob/a416320ccf137b5f842afe7babfb69b46d620d24/src/ts/push-menu.ts#L32-L34

Can the pushmenu of data-pushmenu be written in camel case?

**Reply by danny007in** [reply]
ill change `data-widget` to `data-lte-widget`

**Comment by danny007in** [comment]
> Can AdminLTE v4 retain all Bootstrap css settings?

As pre required

> without modifying the original settings of Bootstrap's css?

Without modifying original settings, Its difficult to built, also it causes duplicate css selectors

**Reply by ZhangChengLin** [reply]
> Without modifying original settings, Its difficult to built, also it causes duplicate css selectors

Bootstrap is just a framework. If you don't want to use some styles, you can avoid the structure of Bootstrap.
Just need to write more css?
